Tarea clase 15

jjuannn opened the pull request.

jjuannn committed:
actualiza

jjuannn committed:
Merge branch 'tarea-clase-13' of https://github.com/jjuannn/pokedex into tarea-clase-13

jjuannn committed:
actualiza

jjuannn committed:
modifica api para uso de entidades

las paginas se guardan en otro modulo

jjuannn committed:
agrega funciones para entidades

agrega funciones para usar clases

jjuannn committed:
modifica las funciones existentes

modifica las funciones ya existentes para poder usar clases

jjuannn committed:
arregla cantidad de pokemones

no mostraba la cantidad de pokemones porque estaba mal modificada la funcion

jjuannn committed:
Revert "modifica api para uso de entidades"

This reverts commit 5f94aa52d26e0fb1bcfc9ca7ac2762f4ea66302c.

jjuannn committed:
Revert "agrega funciones para entidades"

This reverts commit b3f69b89792740796a55e72f819110599929bc69.

jjuannn committed:
Revert "Revert "modifica api para uso de entidades""

This reverts commit df33bf45885dc0c899da36cfc3a11c76a89d44e6.

jjuannn committed:
Revert "Revert "agrega funciones para entidades""

This reverts commit c3c6629dc4dc13b4956a44559e369559dbe68623.

fsodano commented on src/api/api.js:
```diff
@@ -11,21 +11,23 @@ import {
-      return respuesta
+      const respuestaApi = respuestaJSON
+      const primeraPagina = nuevaPagina(respuestaApi)
+      console.log(primeraPagina)
```
esto no hace falta

fsodano commented on src/api/api.js:
```diff
@@ -11,21 +11,23 @@ import {
-      guardarPagina('0', respuestaJSON)
-      const respuesta = respuestaJSON
-      return respuesta
+      const respuestaApi = respuestaJSON
```
fijate que respuestaJSON es el nombre del parámetro que elegiste vos... por qué al nombre del parámetro no le ponés respuestaApi y listo?

fsodano commented on src/api/api.js:
```diff
@@ -11,21 +11,23 @@ import {
+import { nuevaPagina } from '../mapeador/mapeador.js'
 
 export function obtenerInfoPokemones () {
   return fetch('https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20')
```
fijate que estás escribiendo muchas veces la base de la URL, la podés guardar en una `const BASE_URL = ...`

fsodano commented on src/api/api.js:
```diff
@@ -11,21 +11,23 @@ import {
 }
 export function buscarPokemonEnApi(nombre) {
     return fetch(`https://pokeapi.co/api/v2/pokemon/${nombre}`)
       .then(rta => rta.json())
```
respuesta, no abrevies

fsodano commented on src/api/api.js:
```diff
@@ -11,21 +11,23 @@ import {
     return fetch(`https://pokeapi.co/api/v2/pokemon/${nombre}`)
       .then(rta => rta.json())
       .then(rtaJSON => {
-        guardarPokemon(rtaJSON)
         const respuestaPokemon = rtaJSON
         return respuestaPokemon
       })
```
esto no tiene sentido, estás diciendo que vas a tomar un parámetro y lo volvés a devolver así como vino, no hace falta todo este `then`

fsodano commented on src/api/api.js:
```diff
@@ -35,8 +37,7 @@ export function obtenerPaginaSiguienteDesdeAPI () {
     .then(respuesta => respuesta.json())
     .then(respuestaJSON => {
       const { results: pokemones } = respuestaJSON
-      guardarPagina(offset, respuestaJSON)
-      return pokemones
+      return respuestaJSON
     })
```
acá estás tomando la variable pokemones, no hacés nada con esa variable, y devolvés el parámetro original, es decir que todo el 2ndo `then` no hace falta

fsodano replied:
Este error lo tenés en varias partes del código

fsodano commented on src/entidades/paginas.js:
```diff
@@ -0,0 +1,8 @@
+export class Pagina{
+    constructor(respuestaPagina){
```
vos deberías pedir totalPokemones, proximaPagina, paginaSiguiente y listaPokemones y el mapeador es el que se debería encargar de transformar la respuesta de la API en esta entidad. Vos ahora tenés un mapeador que no hace nada y el que se encarga de dividir todo es la entidad, lo tenés al revés

fsodano commented on src/entidades/pokemones.js:
```diff
@@ -0,0 +1,10 @@
+export class Pokemon{
+    constructor(pokemon){
```
mismo comentario que con la página, tomá los atributos que necesitás, esta enttidad no tiene por qué saber que el objeto pokemon tiene `sprites` y `front_default`, esa responsabilidad es del mapper.

fsodano commented on src/entidades/pokemones.js:
```diff
@@ -0,0 +1,10 @@
+    constructor(pokemon){
+        this.nombre = pokemon.name
+        this.altura = pokemon.height
+        this.estadisticas = pokemon.stats
```
stats y types son arrays de objetos, esos objetos también necesitan clases propias que las describan

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+import { obtenerPaginaSiguienteDesdeAPI, obtenerPaginaAnteriorDesdeAPI, buscarPokemonEnApi } from '../api/api.js'
+import { offset } from '../manejador/manejador.js'
+import { buscarPaginaEnLS, buscarPokemonEnLS, guardarPokemon, guardarPagina } from '../storage/storage.js'
```
no abrevies!

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+import { obtenerPaginaSiguienteDesdeAPI, obtenerPaginaAnteriorDesdeAPI, buscarPokemonEnApi } from '../api/api.js'
+import { offset } from '../manejador/manejador.js'
+import { buscarPaginaEnLS, buscarPokemonEnLS, guardarPokemon, guardarPagina } from '../storage/storage.js'
+import { crearListaPokemones } from '../ui/ui.js'
```
tu servicio no tiene que saber de la UI

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+export async function cargarPaginaSiguiente () {
+  try {
+    const paginaLS = buscarPaginaEnLS(offset)
+    if(paginaLS){
+      console.log("cargo ps desde ls")
+    }
```
esto está de más, tenés un try/catch porque buscarPaginaEnLS da un error si no encuentra nada.

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+    return paginaLS.listaPokemones
+  } catch (e) {
+    const respuestaApi = await obtenerPaginaSiguienteDesdeAPI()
+    console.log("cargo desde api")
```
esto está de más

fsodano commented on src/api/api.js:
```diff
@@ -11,21 +11,23 @@ import {
 import {
   offset
 } from '../manejador/manejador.js'
```
🤔 
Tus métodos de la API tienen que tomar el offset como parámetro, no importarlo desde una dependencia

fsodano commented on src/mapeador/mapeador.js:
```diff
@@ -0,0 +1,10 @@
+}
+
+export function nuevaPagina(respuesta){
+    return new Pagina(respuesta)
```
esta función debería cambiar el listado de pokemones por un array de objetos de clase Pokemon

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+    console.log("cargo desde api")
+    const proximaPagina = nuevaPagina(respuestaApi)
+    guardarPagina(offset, proximaPagina)
+    return proximaPagina.listaPokemones
```
por qué cargarPaginaSiguiente devuelve una lista de pokemones en vez de una Pagina?

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+    return proximaPagina.listaPokemones
+  }  
+}
+export async function cargarDataPokemones(nombre){
```
Data, Info, etc. son palabras redundantes, qué estás cargando de los pokemones si no sus datos o información?

fsodano replied:
Esta función debería ser `cargarPokemon` porque estás cargando 1 solo.

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+  let pokemon
+  pokemon = buscarPokemonEnLS(nombre)
+  if(pokemon){
+    console.log("lo cargo desde ls")
```
los console.log son para testear pero no deberían formar parte del codebase generalmente

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+    pokemon = nuevoPokemon(respuestaApi)
+    guardarPokemon(pokemon)
+    console.log("lo cargo desde api")
+    return pokemon
```
podés tener un único `return` por fuera del `if`, para eso te sirve haber declarado el `let pokemon` fuera del if

fsodano commented on src/servicios/servicios.js:
```diff
@@ -0,0 +1,49 @@
+    console.log("lo cargo desde ls")
+    return pokemon
+  } else {
+    let respuestaApi = await buscarPokemonEnApi(nombre)
```
```suggestion
    const respuestaApi = await buscarPokemonEnApi(nombre)
```

fsodano commented on src/ui/ui.js:
```diff
@@ -54,17 +54,17 @@ function mostrarInfoPokemones (peso, altura, nombre, vida, ataque, defensa, ataq
 }
 
 export function actualizarInformacion(rtaJSON) {
```
en vez de un rtaJSON esto debería ser un `pokemon`

fsodano commented on src/ui/ui.js:
```diff
@@ -54,17 +54,17 @@ function mostrarInfoPokemones (peso, altura, nombre, vida, ataque, defensa, ataq
+  const altura = rtaJSON.altura
+  const nombre = rtaJSON.nombre
+  const $foto = rtaJSON.fotos
+  const hp = Object.values(rtaJSON.estadisticas[0])[0]
+  const ataque = Object.values(rtaJSON.estadisticas[1])[0]
+  const defensa = Object.values(rtaJSON.estadisticas[2])[0]
+  const ataqueEspecial = Object.values(rtaJSON.estadisticas[3])[0]
+  const defensaEspecial = Object.values(rtaJSON.estadisticas[4])[0]
+  const velocidad = Object.values(rtaJSON.estadisticas[5])[0]
```
todo esto no le corresponde a la UI, todo esto debería ya estar acomodado en objetos, listo para que la ui lo consuma!

fsodano commented on src/ui/ui.js:
```diff
@@ -54,17 +54,17 @@ function mostrarInfoPokemones (peso, altura, nombre, vida, ataque, defensa, ataq
+  const peso = rtaJSON.peso
+  const altura = rtaJSON.altura
+  const nombre = rtaJSON.nombre
+  const $foto = rtaJSON.fotos
```
por qué esto tiene un `$`? Y por qué si es 1 sola foto estás pidiendo el atributo `fotos`?

fsodano reviewed (changes requested):
Te dejé varios comentarios, fijate y me avisás.

Una cosa más es que `mostrarInfoPokemones` muestra la info de 1 solo pokemon, por lo cual debería ser `mostrarPokemon` (el info, data, etc. siempre están de más)

jjuannn committed:
Update src/servicios/servicios.js

Co-authored-by: Fabricio Leonardo Sodano Pascazi <fabricio.sodano@gmail.com>

jjuannn committed:
agrega url guardada en una const

agrega la const baseURL y las usa en los fetch para no escribir muchas veces la misma URL

jjuannn committed:
actualiza la clase pagina

actualiza la clase pagina, borra parametros que no le correspondian y los corrige

jjuannn committed:
actualiza la clase pokemon

actualiza la clase pokemon. Ahora podemos acceder a las estadisticas del pokemon directamente

jjuannn committed:
borra comentario

borra comentario

jjuannn committed:
actualiza mapeador

actualiza mapeador para aplicar las correciones de pokemon.js y pagina.js

jjuannn committed:
actualiza el modulo servicios

Borra lineas innecesarias y console.log. tambien borra la funcion que obtenia la pagina anterior desde api, ya que si cuando cargamos una pagina la guardamos en local storage, esta funcion era innecesaria

jjuannn committed:
actualiza nombre de las funciones

cambia los nombres de las funciones y deja de usar abreviaturas

jjuannn committed:
Arregla errores varios

cambia las constantes dentro de actualizarInformacion, para que la actualize con lo que le devuelve el mapeador.
